edalize/tools/yosys.py

```diff
@@ -131,9 +135,65 @@ def write_config_files(self, edam):
             )
 
         commands = EdaCommands()
-        commands.add(
-            ["yosys", "-l", "yosys.log", "-p", f"'tcl {template}'"],
-            [default_target],
-            [template] + depfiles,
-        )
+
+        # First, check if split_io list is passed in and is the correct size
+        split_io_list = []
+        if (
+            self.tool_options.get("split_io")
+            and len(self.tool_options.get("split_io")) == 8
+        ):
+            split_io_list = self.tool_options.get("split_io")
+
+        # Configure first call to Yosys
+        targets = []
+        depends = depfiles
+        variables = []
+
+        if split_io_list:
+            targets = f"{self.name}.json"
+            variables = {
+                "USE_ROI": "FALSE",
+                "TECHMAP_PATH": "$F4PGA_SHARE_DIR/techmaps/xc7_vpr/techmap",
+                "TOP": self.toplevel,
+                "INPUT_XDC_FILES": split_io_list[0],
+                "PART_JSON": split_io_list[1],
+                "OUT_FASM_EXTRA": f"{self.name}_fasm_extra.fasm",
+                "OUT_SDC": f"{self.name}.sdc",
+                "OUT_SYNTH_V": f"{self.name}_synth.v",
+                "PYTHON3": "$(which python3)",
+                "UTILS_PATH": "$F4PGA_SHARE_DIR/scripts",
+                "OUT_JSON": targets,
+            }
+        else:
+            targets = [default_target]
+            depends = [template] + depends
+
+        command = ["yosys", "-l yosys.log", f"-p 'tcl {template}'"]
+
+        if split_io_list:
+            command += depfiles
+
+        commands.add(command, targets, depends, variables=variables)
+
+        # Configure python script and additional call to Yosys
+        if split_io_list:
```

Comment:
FYI, `conv.tcl` and executing `split_inouts` were merged into the `synth.tcl` script. A single call to yosys is enough now. See https://github.com/chipsalliance/f4pga/pull/633.

Reply:
Oh, interesting, I will have to update my f4pga and take a look at that

Reply:
@umarcor [ed9646c](https://github.com/olofk/edalize/pull/335/commits/ed9646cd6ee45a1da9dbc19d89994d2163c3cf00) should now reflect the f4pga change